crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -327,6 +327,15 @@ def prepare_nodes_for_os_upgrade
       node.save
     end
 
+    # change rabbitmq/drbd setting to native queues setup (new default)
+    prop = Proposal.find_by(barclamp: "rabbitmq", name: "default")
+    role = prop.role
+    if role.default_attributes["rabbitmq"]["ha"]["storage"]["mode"] == "drbd"
+      role.default_attributes["rabbitmq"]["cluster"] = true
+      role.default_attributes["rabbitmq"]["erlang_cookie"] = random_password
+      role.save
+    end
```

Comment:
I think you need to update both, the proposal and the role. Otherwise the change will get lost if the user clicks apply on the proposal again.